photutils/psf/psfphotometry.py

```diff
@@ -0,0 +1,446 @@
+            return self._aperture_radius
+
+        @niters.setter
+        def aperture_radius(self, aperture_radius):
+            if isinstance(aperture_radius, (int, float)) and aperture_radius > 0:
+                self._aperture_radius = aperture_radius
+            else:
+                raise ValueError('aperture_radius must be a real-valued '
+                                 'number, received aperture_radius = {}'
+                                 .format(aperture_radius))
+
+
+    def __call__(self, image, positions=None):
+        """
+        Parameters
+        ----------
+        image : 2D array-like, `~astropy.io.fits.ImageHDU`,
+        `~astropy.io.fits.HDUList`
+            Image to perform photometry.
+        positions : `~astropy.table.Table` (optional)
+            Positions, in pixel coordinates, at which stars are located.
+            The columns must be named as 'x_0' and 'y_0'. 'flux_0' can also
+            be provided to set initial fluxes.
+
+        Returns
+        -------
+        outtab : `~astropy.table.Table`
+            Table with the photometry results, i.e., centroids and fluxes
+            estimations.
+        residual_image : array-like, `~astropy.io.fits.ImageHDU`,
+        `~astropy.io.fits.HDUList`
+            Residual image calculated by subtracting the fitted sources
+            and the original image.
+        """
+        
+        if positions is None:
+            return self.do_photometry(image)
+        else:
+            return self.do_fixed_photometry(image, positions)
+
+    def do_photometry(self, image):
+        """
+        Parameters
+        ----------
+        image : 2D array-like, `~astropy.io.fits.ImageHDU`,
+        `~astropy.io.fits.HDUList`
+            Image to perform photometry
+        
+        Returns
+        -------
+        outtab : `~astropy.table.Table`
+            Table with the photometry results, i.e., centroids and fluxes
+            estimations.
+        residual_image : array-like, `~astropy.io.fits.ImageHDU`,
+        `~astropy.io.fits.HDUList`
+            Residual image calculated by subtracting the fitted sources
+            and the original image.
+        """
+
+        outtab = Table([[], [], [], [], [], []],
+                       names=('id', 'group_id', 'x_fit', 'y_fit', 'flux_fit',
+                              'iter_detected'),
+                       dtype=('i4', 'i4', 'f8', 'f8', 'f8', 'i4'))
+
+        residual_image = image.copy()
+        residual_image = residual_image - self.bkg(image)
+        sources = self.find(residual_image)
+        
+        if self.aperture_radius is None:
+            self.aperture_radius = self.psf.sigma.value*gaussian_sigma_to_fwhm
+
+        apertures = CircularAperture((sources['xcentroid'],
+                                      sources['ycentroid']),
+                                     r=self.aperture_radius)
+
+        sources['flux'] = aperture_photometry(residual_image,
```

Comment:
very minor, but you might call this "aperture_flux" or similar to distinguish from the psf-fitted flux.  this table is only used internally so that's not a big deal, but it might be useful for debugging or others who want to hack on this in the future or the like.